### PR TITLE
Added a decorator

### PR DESCRIPTION
**Pull request by RHagenaars**
I've added a decorator for conveniently defining jobs from functions.

### CODE REVIEW

**Comment by Naramsim** [comment]
Can you write pls how to use this decorator?

Thanks

**Comment by dbader** [comment]
Thanks RHagenaars for creating this PR. To tell you the truth I'm on the fence whether this should be a part of schedule itself or an FAQ item.

A couple of thoughts here:

* Please use `functools.wraps` to ensure the metadata of the wrapped function is copied over.
* The code example doesn't import `repeat()`
* I don't think the code example will render correctly in the HTML docs, this would also need fixing.

**Comment by RHagenaars** [comment]
@Naramsim the usage would be as follows:

```
@repeat(every(10).minutes)
def f():
  print('This function is repeated.')
```

@dbader thanks for taking time to consider it. About your thoughts:

> Please use functools.wraps to ensure the metadata of the wrapped function is copied over.

That's not necessary in this case. If you look closely, you'll see that the actual decorated function is returned and not some wrapper.

> The code example doesn't import repeat()

Good catch! I've added that import.

> I don't think the code example will render correctly in the HTML docs, this would also need fixing.

I'm not sure how to fix that. Could you elaborate some more on what exactly needs to be fixed? I took the docstring on top of the file for an example.

> I'm on the fence whether this should be a part of schedule itself or an FAQ item

That's up to you of course. I personally like decorators for things like this, because it allows you to put coupled statements - the function and the schedule - close to each other. I think that in most cases, the code is more clear when these are not separated.

**Comment by SijmenHuizenga** [comment]
Hi @RHagenaars, Thanks a lot for submitting this pr! I'm thrilled to add this to the library :100:

Just now I've added support for passing arguments to the function via the decorator. And I've added some examples to the docs. Would you like to let me know what you think of this?

File: docs/examples.rst [hunk 1]
```diff
@@ -43,10 +43,30 @@ Run a job every x minute
...
     while True:
-        # run_pending
         schedule.run_pending()
         time.sleep(1)
 
+Use a decorator to schedule a job
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``@repeat`` to schedule a function.
+Pass it an interval using the same syntax as above while omitting the ``.do()`` with.
```
Comment (RHagenaars) [inline docs/examples.rst:53]: That line doesn't read very well:
```suggestion
Pass it an interval using the same syntax as above while omitting the ``.do()``.
```

File: docs/examples.rst [hunk 2]
```diff
@@ -43,10 +43,30 @@ Run a job every x minute
...
+
+    @repeat(every(10).minutes)
+    def job():
+        print("I am a scheduled job")
+
+    while True:
+        run_pending()
+        time.sleep(1)
+
+The ``@every`` decorator does not work on non-static class methods.
```
Comment (RHagenaars) [inline docs/examples.rst:68]:
```suggestion
The ``@repeat`` decorator does not work on non-static class methods.
```

**Review by RHagenaars — COMMENTED** [review]
Looks good to me, @SijmenHuizenga ! I really like the addition of passing arguments.

**Comment by SijmenHuizenga** [comment]
Thanks for the review @RHagenaars! Let's merge :rocket:

**Pull request merged**